torch/distributed/_spmd/distribute.py

```diff
@@ -740,11 +738,47 @@ def _convert_to_distributed(
                         output_schemas[inp_arg.name] = Schema(
                             obj.device_mesh, obj.placements  # type: ignore[arg-type]
                         )
-
         elif node.op == OP.CALL_FUNCTION:
             args = tree_map(partial(_remap_arg, node_to_obj), node.args)
             kwargs = tree_map(partial(_remap_arg, node_to_obj), node.kwargs)
-            node_to_obj[node] = node.target(*args, **kwargs)
+
+            dsymints = list(
+                filter(lambda a: isinstance(a, DSymInt), args + tuple(kwargs.values()))
+            )
+
+            if node.target in operators and len(dsymints) > 0:
+                assert all(
+                    dsymints[0].mesh == d.mesh for d in dsymints
+                ), "all DSymInts must have the same mesh. "
+
+                local_args = tree_map(
```

Comment:
nit: these mapping can be simplified with `tree_map_only`
```
local_args = tree_map_only(DSymInt, lambda a: a.local_value, args)
```

Reply:
Thanks! Will fix in a PR on top